Modules/Core/Common/include/itkImageBase.h

```diff
@@ -465,8 +465,8 @@ class ITK_TEMPLATE_EXPORT ImageBase : public DataObject
    *
    * \sa Transform */
   template <typename TCoordRep>
-  [[nodiscard]] bool
-  TransformPhysicalPointToIndex(const Point<TCoordRep, VImageDimension> & point, IndexType & index) const
+  ITK_NO_DISCARD(Call the overload which has the point as the only parameter and returns the index)
+  bool TransformPhysicalPointToIndex(const Point<TCoordRep, VImageDimension> & point, IndexType & index) const
```

Comment:
Please consider to leave the unconditional `[[nodiscard]]` (even for legacy-only) here. For `TransformPhysicalPointToIndex`, it's really a _no-brainer_ to replace it with the other overload. (It does not have the `template` difficulty that TransformPhysicalPointToContinuousIndex has.)

Reply:
Even if replacement is easy, it requires code changes. The legacy flag's purpose is to allow this to be made gradually.